src/Symfony/Component/Serializer/Context/Encoder/YamlEncoderContextBuilder.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Context\Encoder;
+
+use Symfony\Component\Serializer\Context\ContextBuilder;
+use Symfony\Component\Serializer\Encoder\YamlEncoder;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+
+/**
+ * A helper providing autocompletion for available YamlEncoder options.
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ */
+final class YamlEncoderContextBuilder extends ContextBuilder
+{
+    /**
+     * Configures the threshold to switch to inline YAML.
+     */
+    public function withInlineThreshold(?int $inlineThreshold): static
+    {
+        return $this->with(YamlEncoder::YAML_INLINE, $inlineThreshold);
+    }
+
+    /**
+     * Configures the indentation level.
+     *
+     * Must be positive.
+     *
+     * @throws InvalidArgumentException
+     */
+    public function withIndentLevel(?int $indentLevel): static
+    {
+        if (null !== $indentLevel && $indentLevel < 0) {
```

Comment:
Maybe could you use the appropriate Psalm type instead of a hard check? https://psalm.dev/docs/annotating_code/type_syntax/scalar_types/#positive-int